addons/mail/static/src/messaging/component/activity_mark_done_popover/activity_mark_done_popover_tests.js

```diff
@@ -0,0 +1,285 @@
+        activity_category: 'not_upload_file',
+        can_write: true,
+        id:12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+    document.addEventListener('o-discard-clicked', () => {
+        assert.step('event_triggered');
+    }, true);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        'Popover component should be present'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        'Popover component should contain the discard button'
+    );
+    document.querySelector('.o_ActivityMarkDonePopover_discardButton').click();
+    assert.verifySteps(['event_triggered'], 'Discard clicked should trigger the right event');
+
+});
+
+QUnit.test('activity mark done popover mark done without feedback', async function (assert) {
+    assert.expect(7);
+
+    await this.start({
+        debug: true,
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/action_feedback') {
+                assert.step('action_feedback');
+                assert.strictEqual(args.args.length, 1);
+                assert.strictEqual(args.args[0].length, 1);
+                assert.strictEqual(args.args[0][0], 12);
+                assert.strictEqual(args.kwargs.attachment_ids.length, 0);
+                assert.notOk(args.kwargs.feedback);
+                return;
+            }
+            if (route === '/web/dataset/call_kw/mail.activity/unlink') {
+                // should not be called
+                assert.step('unlink');
+            }
+            return this._super(...arguments);
+        },
+    });
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        id:12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+
+    document.querySelector('.o_ActivityMarkDonePopover_doneButton').click();
+    assert.verifySteps(
+        ['action_feedback'],
+        "Mark done and schedule next button should call the right rpc"
+    );
+});
+
+QUnit.test('activity mark done popover mark done with feedback', async function (assert) {
+    assert.expect(7);
+
+    await this.start({
+        debug: true,
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/action_feedback') {
+                assert.step('action_feedback');
+                assert.strictEqual(args.args.length, 1);
+                assert.strictEqual(args.args[0].length, 1);
+                assert.strictEqual(args.args[0][0], 12);
+                assert.strictEqual(args.kwargs.attachment_ids.length, 0);
+                assert.strictEqual(args.kwargs.feedback, 'This task is done');
+                return;
+            }
+            if (route === '/web/dataset/call_kw/mail.activity/unlink') {
+                // should not be called
```

Comment:
Same question, and do we actually need to verify this?
How do we decide all the routes that we want to check for not being called?

Reply:
See other comment, it cames to be because the unlink was called through `activity.delete()` after marking it as done (which already "delete" it from server) so the RPC was failing. So here we check if the unlink is still made or not.